docs/libcurl/opts/CURLOPT_HTTPTRAILER_FUNCTION.3

```diff
@@ -0,0 +1,102 @@
+.\" **************************************************************************
+.\" *                                  _   _ ____  _
+.\" *  Project                     ___| | | |  _ \| |
+.\" *                             / __| | | | |_) | |
+.\" *                            | (__| |_| |  _ <| |___
+.\" *                             \___|\___/|_| \_\_____|
+.\" *
+.\" * Copyright (C) 1998 - 2017, Daniel Stenberg, <daniel@haxx.se>, et al.
+.\" *
+.\" * This software is licensed as described in the file COPYING, which
+.\" * you should have received as part of this distribution. The terms
+.\" * are also available at https://curl.haxx.se/docs/copyright.html.
+.\" *
+.\" * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+.\" * copies of the Software, and permit persons to whom the Software is
+.\" * furnished to do so, under the terms of the COPYING file.
+.\" *
+.\" * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+.\" * KIND, either express or implied.
+.\" *
+.\" **************************************************************************
+.\"
+.TH CURLOPT_HTTPTRAILER_FUNCTION 3 "14 Aug 2018" "libcurl 7.61.0" "curl_easy_setopt options"
+
+.SH NAME:
+CURLOPT_HTTPTRAILER_FUNCTION \- Callback to fill trailing headers list.
+		
+.SH SYNOPSIS:
+#include <curl.h>
+
+int curl_trailer_callback(struct curl_slist ** list, void *userdata); 
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_HTTPTRAILER_FUNCTION, curl_trailer_callback *func);
+
+.SH DESCRIPTION:
+Pass a pointer to a callback function.
+
+This callback function will be called right before sending the final CR LF 
+in an HTTP transfer to fill a list of trailing headers to be sent before finishing the
+HTTP transfer.
+
+The callback must return the number of traling headers put in the list.
+Set the userdata argument with the CURLOPT_HTTPTRAILER_DATA option.
```

Comment:
what happens if it doesn't? How can the callback return an error that stops everything and bails out? (if for example the code runs into a non-recoverable error such as out of memory or similar)

Reply:
This remains from a previous iteration where the knowledge of the number of trailers was crucial, in fact, with this implementation, the callback should now return `void`.  
I still have to work around the fact that `Curl_sendbuffer` and associated facilities are not available if curl isn't built with HTTP, so the next iteration may actually need this information, if not, I will make the callback a `void` and adjust the documentation.